pkg/collector/remoteconfig/remote_config.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package remoteconfig schedule and unschedule integrations through remote-config
+// using the AGENT_INTEGRATIONS product
+package remoteconfig
+
+import (
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/secrets"
+	pkglog "github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Scheduler is the structure used to receive remote-config checks, and add or remove
+// them in the agent's scheduler
+type Scheduler struct {
+	scheduler     *collector.CheckScheduler
+	runningChecks []integration.Config
+}
+
+type agentIntegration struct {
+	Name       string            `json:"name"`
+	Instances  []json.RawMessage `json:"instances"`
+	InitConfig json.RawMessage   `json:"init_config"`
+}
+
+// secretsDecrypt allows tests to intercept calls to secrets.Decrypt.
+var secretsDecrypt = secrets.Decrypt
+
+// NewScheduler creates an instance of a remote config integration scheduler
+func NewScheduler(scheduler *collector.CheckScheduler) *Scheduler {
+	return &Scheduler{
+		runningChecks: make([]integration.Config, 0),
+		scheduler:     scheduler,
+	}
+}
+
+// IntegrationScheduleCallback is called at every AGENT_INTEGRATIONS to schedule/unschedule integrations
+func (sc *Scheduler) IntegrationScheduleCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	if sc.scheduler == nil {
+		pkglog.Debugf("Scheduler is not ready to start remote-config integrations")
+		return
+	}
+
+	// Unschedule every integrations, even if they haven't changed
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏It this safe? Won't that reset the integrations check schedule? If for instance an integration checks every hour (idk if possible), could we lose some points?